### DISCUSSION TITLE
Approval workflow with hierarchical data

### DISCUSSION THREAD

**Discussion by mubukp** [elsa-workflows/elsa-core#2600] [Q&A] [answered]
What is the best way to create a hierarchical approval in Elsa? For example a request to be approved by initiators manager and then his manager etc. 

In addition, how do we create a "my to do" view and current and future approver views?

**Comment by sfmskywalker** [comment, marked as answer]
That's a tough question to answer in a few sentences. I would start with modelling the request, which perhaps carries the list of managers that the workflow can iterate on using the **For Each** activity to send each manager an email for example.
There are many ways to go about this, and it all depends on the details.

Same goes with regards to the question of creating a "my to do" view. This view could be based on stored bookmarks in the database (where a bookmark represents a blocking activity waiting for a manager's approval), but you could also create TodoItems in a table in response to events happening within the workflow.

As a general guide, I recommend to start simple and make it more complex only when necessary.

**Reply by mubukp** [reply]
Hello,

Thanks for the input and it worked well. 

However, we have a scenario wherein you can return the request to the initiator for corrections. Is there a way where we can return a workflow instance to the start or to a specific activity?

**Reply by sfmskywalker** [reply]
Yes. The typical way to do so is to simply connect some activity to a specific activity. For example, let's say you have an activity that sends a request for review to some reviewer, who then determined that it needs to be sent back to the initiator for corrections. The workflow would be blocked on an activity that represents this stage, like a Signal Received activity. You could have two of them: one waiting for the "approved" signal and another one waiting for the "request corrections" signal. The latter activity would simply have to be connected back to some other activity on your workflow that represents asking the initiator for corrections.
